debian/Dockerfile

```diff
@@ -34,16 +34,18 @@ RUN set -ex && \
       apt-key adv --no-tty --batch --keyserver hkp://keyserver.pgp.com:80 --recv-keys "$key" || \
       apt-key adv --no-tty --batch --keyserver hkp://ipv4.pool.sks-keyservers.net:80 --recv-keys "$key" ; \
     done && \
-    echo "deb https://apt.dockerproject.org/repo debian-jessie main" > /etc/apt/sources.list.d/docker.list && \
+    curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add - && \
+    echo "deb https://download.docker.com/linux/debian buster stable" > /etc/apt/sources.list.d/docker.list && \
     apt-get update && \
-    apt-get install -y docker-engine && \
+    apt-get install -y docker-ce docker-ce-cli && \
```

Comment:
Do we need all of `docker-ce` or is `docker-ce-cli` good enough?

Reply:
`docker-ce-cli` give us `/usr/bin/docker`, shell autocompletion, and manpages, while `docker-ce` actually handles things like the init/conf scripts and docker binaries.  If these are just for jenkins-agents, we might be able to live with the cli only.

```
# dpkg -L docker-ce-cli
/.
/usr
/usr/bin
/usr/bin/docker
/usr/libexec
/usr/libexec/docker
/usr/libexec/docker/cli-plugins
/usr/libexec/docker/cli-plugins/docker-app
/usr/libexec/docker/cli-plugins/docker-buildx
... man pages and the likes
```

```
# dpkg -L docker-ce
/.
/etc
/etc/default
/etc/default/docker
/etc/init
/etc/init/docker.conf
/etc/init.d
/etc/init.d/docker
/lib
/lib/systemd
/lib/systemd/system
/lib/systemd/system/docker.service
/lib/systemd/system/docker.socket
/usr
/usr/bin
/usr/bin/docker-init
/usr/bin/docker-proxy
/usr/bin/dockerd
/usr/share
/usr/share/doc
/usr/share/doc/docker-ce
/usr/share/doc/docker-ce/README.md
/usr/share/doc/docker-ce/changelog.Debian.gz
/var
/var/lib
/var/lib/docker-engine
/var/lib/docker-engine/distribution_based_engine.json
```

Reply:
Honestly I don't really remember if we're using Docker in these images or not. I kind of think not, because I may have added it to try to get privileged Docker-in-Docker builds working, and we've moved away from that. Unless someone can think of why we need it, I'd be comfortable leaving it out entirely in this iteration—we can put it back if it turns out I'm forgetting something.

Reply:
Updated to remove docker.